.eslintrc.js

```diff
@@ -5,12 +5,15 @@ module.exports = {
     server: true,
   },
   root: true,
-  parser: 'babel-eslint',
+  parser: '@babel/eslint-parser',
   parserOptions: {
-    ecmaVersion: 2018,
+    ecmaVersion: 'latest',
```

Comment:
When running `yarn lint:js` locally, there are 777 errors, 625 are potentially fixable with `--fix`

Reply:
This is a good time to decide on an approach for linting across all of our Ember apps as they make their way to 4.12. I'll try to summarize my suggestions here:

1. As part of the move to 4.12, we should get any repos that use the deprecated `eslint-plugin-ember-best-practices` package off of it. This repo already doesn't use it (hence the stricter linting) but this would apply to my downstream 4.12 Zola branch. This will help to get all of our Ember repos on even footing.
2. We should apply and commit any changes that the linter can make on it's own with `--fix`, ideally in their own commit. This seems like a no brainer to me.
3. Looking through the errors I get after running `--fix` on this repo, it looks like they all either come from `ember-plugin-eslint` or `eslint-plugin-qunit`. To describe the approach I think we should take, I'll bucket errors into those that come from `ember-plugin-eslint` and those that don't:
    a. Errors that don't come from `ember-plugin-eslint`: We should default to resolving these as a part of the Ember 4.12 work. If we encounter any such errors in other projects that we determine are going to be harder to fix than they're worth, we can tweak `.eslintrc.js` files to change the relevant rules to warns on a case-by-case basis
    b. Errors that do come from `ember-plugin-eslint`: It looks like a sizable chunk of these errors that can't be fixed automatically are recommending that we change away from Ember APIs that are deprecated but not yet actually removed. In particular, a lot of them refer to changing away from the "classic" component syntax to Glimmer components. Obviously this would be great to do but our approach to these upgrades has been, and should continue to be, about getting to supported version of Node and Ember. Doing so doesn't require us to get off of deprecated APIs. For those reasons, I suggest we update the `eslintrc.js` file to mark rules related to deprecated features as `warn` instead of `error`. For rules that _don't_ appear to be related to deprecated APIs, the developer should make an effort to resolve them. It will be up to the developer to determine if a rule is related to a deprecated API and their changes to `eslintrc.js` will be subject to PR code review just like anything else. I'll update the eslintrc.js in this PR accordingly and that config can serve as a "starting point" to be used in other repos as we upgrade them to 4.12. However, more rule changes may need to be added to other repos if they happen to have code that violates rules that this repo doesn't. We could even publish our own Ember 4.12-specific eslint to import into repos with `extends` if we want to truly centralize the configuration. When all is said and done, we will be able to lint our Ember repos and consider the output "binding" without having to make changes for APIs that are deprecated but not removed (with eslint printing warnings for those rules).

A note about prettier - I suggest we stick with whatever prettier configs are spat out by `ember-cli-update` and fix the code accordingly. This should ensure consistent prettier configuration across all Ember repos.